nixos/modules/services/web-apps/rundeck.nix

```diff
@@ -0,0 +1,550 @@
+        example = lib.literalExpression ''
+          {
+            "admin.aclpolicy" = '''
+              description: Admin access for administrators
+              context:
+                project: '.*'
+              for:
+                resource:
+                  - allow: '*'
+                job:
+                  - allow: '*'
+                node:
+                  - allow: '*'
+              by:
+                group: admin
+              ---
+              description: Admin access in application scope
+              context:
+                application: 'rundeck'
+              for:
+                resource:
+                  - allow: '*'
+                project:
+                  - allow: '*'
+              by:
+                group: admin
+            ''';
+          }
+        '';
+      };
+
+      extraConfig = lib.mkOption {
+        type = lib.types.attrs;
+        default = { };
+        description = "Additional configuration options for rundeck-config.properties";
+        example = lib.literalExpression ''
+          {
+            "rundeck.gui.title" = "My Rundeck Server";
+            "rundeck.feature.repository.enabled" = "true";
+            "rundeck.projectsStorageType" = "db";
+          }
+        '';
+      };
+
+      database = {
+        type = lib.mkOption {
+          type = lib.types.enum [
+            "h2"
+            "postgresql"
+            "mysql"
+          ];
+          default = "h2";
+          description = "Database type to use (h2, postgresql, or mysql)";
+        };
+
+        host = lib.mkOption {
+          type = lib.types.str;
+          default = "localhost";
+          description = "Database host";
+        };
+
+        port = lib.mkOption {
+          type = lib.types.port;
+          default = 5432;
+          description = "Database port";
+        };
+
+        name = lib.mkOption {
+          type = lib.types.str;
+          default = "rundeck";
+          description = "Database name";
+        };
+
+        username = lib.mkOption {
+          type = lib.types.str;
+          default = "rundeck";
+          description = "Database username";
+        };
+
+        password = lib.mkOption {
+          type = lib.types.str;
+          default = "rundeck";
+          description = "Database password";
+        };
+      };
+
+      settings = lib.mkOption {
+        type = lib.types.attrsOf (
+          lib.types.oneOf [
+            lib.types.bool
+            lib.types.int
+            lib.types.str
+          ]
+        );
+        default = { };
+        description = ''
+          Rundeck configuration options.
+          See https://docs.rundeck.com/docs/administration/configuration/config-file-reference.html
+          for all available options.
+        '';
+        example = lib.literalExpression ''
+          {
+            "server.address" = "0.0.0.0";
+            "server.port" = "4440";
```

Comment:
Those are all duplicated with the above options. Also we should mention how they interact with each other

Reply:
I heard I'll look at how to do better.